docs/book/src/clusterctl/commands/upgrade.md

```diff
@@ -75,6 +77,26 @@ clusterctl upgrade apply \
 
 <aside class="note warning">
 
+<h1>Clusterctl upgrade test coverage</h1>
+
+Cluster API only tests a subset of possible clusterctl upgrade paths as otherwise the test matrix would be overwhelming.
+Untested upgrade paths are not blocked by clusterctl and should work in general, they are just not tested. Users 
```

Comment:
```suggestion
Untested upgrade paths are not blocked by clusterctl but they are just not tested. Users 
```

Nit, but I'm not sure about the grey area of "it should work, but it's not tested". Better IMO just to describe the state.

Reply:
Let's see what others thing. I think the assurance that this is something that we expect to work out-of-the-box might be important to users.

Reply:
We can keep it as it is, given that the most important part is the next part about users doing their own validation if they want to rely on untested upgrade paths